@3rdweb-sdk/react/hooks/useLoggedInUser.ts

```diff
@@ -2,11 +2,20 @@ import { useAddress, useUser } from "@thirdweb-dev/react";
 
 export function useLoggedInUser(): ReturnType<typeof useUser> {
   const connectedAddress = useAddress();
-  const user = useUser();
-  // ONLY RETURN THE USER IF THE ADDRESS MATCHES
-  if (user.user?.address === connectedAddress) {
-    return user;
+  const userQuery = useUser();
+
+  // user is not considered logged in if the connected address does not match the user's address
+  if (
+    userQuery.isLoggedIn &&
+    connectedAddress &&
+    userQuery.user?.address !== connectedAddress
+  ) {
+    return {
+      user: undefined,
+      isLoading: userQuery.isLoading,
+      isLoggedIn: false,
+    };
   }
 
-  return { user: undefined, isLoading: user.isLoading, isLoggedIn: false };
+  return userQuery;
```

Comment:
@jnsdls we updated this one with @MananTank 

fixes the issue with tracking isLoading and isLoggedIn state. F.ex:

```
if (!isLoading && !isLoggedIn) {
  return  <Connect />
}

return <Component />
```

does it look good to you?